xarray/core/dataarray.py

```diff
@@ -1705,7 +1705,8 @@ def stack(
         ----------
         dimensions : Mapping of the form new_name=(dim1, dim2, ...)
             Names of new dimensions, and the existing dimensions that they
-            replace.
+            replace. Passing a list containing an ellipsis (`[...]`)
+            dimensions will stack over all dimensions.
```

Comment:
`Passing a partial list of dimensions with ellipsis is also supported ([..., dim2]).`

Reply:
What do you think of this? I wanted to say what the ellipsis does. Or do you think unneeded / people can try it and will work it out.

Reply:
I think we should always keep the docstrings exhaustive since it makes it easier for people.

Also think both DataArray and Dataset should have the same two sentences:

 An ellipsis (`...`) will be replaced by all unlisted dimensions.Passing a list containing an ellipsis (`stacked_dim=[...]`)  will stack over all dimensions.

Reply:
Very good!